core/chains/evm/logpoller/orm.go

```diff
@@ -320,27 +317,37 @@ func (o *DSORM) DeleteBlocksBefore(ctx context.Context, end int64, limit int64)
 	}
 
 	// Remove up to limit blocks at a time, until we've reached the limit or removed everything eligible for deletion
-	var deleted, rows int64
-	for limitBlock += (limit - 1); deleted < limit; limitBlock += limit {
-		if limitBlock > end {
-			limitBlock = end
-		}
-		result, err = o.ds.ExecContext(ctx, `DELETE FROM evm.log_poller_blocks WHERE block_number <= $1 AND evm_chain_id = $2`, limitBlock, ubig.New(o.chainID))
-		if err != nil {
-			return deleted, err
+	var upper int64
+	for lower := start; numResults < limit; lower = upper + 1 {
+		upper = lower + limit - 1
+		if upper > end {
+			upper = end
 		}
-
-		if rows, err = result.RowsAffected(); err != nil {
-			return deleted, err
+		rows, err2 := query(lower, upper)
```

Comment:
Here you return `err` that is nil instead of `err2`. Is it desired? 
Is it not cleaner and less error-prone to reuse err instead of defining a new err2 variable?

Reply:
Good catch, thanks!